### ISSUE TITLE
Matches returned on filepath for text search query

### ISSUE THREAD

**Issue by keegancsmith** [google/zoekt#46]
Reproduction on public instance. This should have no matches, but instead it is returning a single match for line 0, but that is not the line.

https://cs.bazel.build/search?q=mapreduce.*raw+r%3Ahadoop++f%3ARawKeyValueIterator&num=50

![image](https://user-images.githubusercontent.com/187831/35910707-2288dff2-0c00-11e8-8304-122f3fa1f247.png)

**Comment by hanwen** [comment]
this is working as intended, albeit not documented. 
 
  "str"

gets expanded to 

 (OR file:"str" content:"str")

if you want to search for content specifically, you can use 

  content:mapreduce

This has two reasons:

* better search quality
* it creates a place for highlighting matches on the filename.

I'm open to suggestions on how to make this more obvious.

**Comment by keegancsmith** [comment]
Yeah that makes a lot of sense. I think the bit that is surprising to me is how it is displayed in the UI, where it looks like another line match.

**Comment by hanwen** [comment]
Happy for a patch that changes the display too. Maybe we could do match highlighting on filenames in general.
